### ISSUE TITLE
Error found with fuzzing

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/area-builder#71]
```
Error: Assertion failed: Error: Specified shape to be replaced does not appear to be present.
    at window.assertions.assertFunction (http://192.168.0.2:8080/assert/js/assert.js:22:13)
    at ShapePlacementBoard.inherit.replaceShapeWithUnitSquares (http://192.168.0.2:8080/area-builder/js/common/model/ShapePlacementBoard.js?bust=1423077783575:905:45)
    at decomposeShape (http://192.168.0.2:8080/area-builder/js/game/model/AreaBuilderGameModel.js?bust=1423077783575:88:44)
    at http://192.168.0.2:8080/area-builder/js/game/model/AreaBuilderGameModel.js?bust=1423077783575:91:69
    at Array.wrapper (http://192.168.0.2:8080/axon/js/Property.js?bust=1423077783575:224:11)
    at Property.inherit._notifyObservers (http://192.168.0.2:8080/axon/js/Property.js?bust=1423077783575:103:29)
    at Property.inherit._setAndNotifyObservers (http://192.168.0.2:8080/axon/js/Property.js?bust=1423077783575:89:14)
    at Property.inherit.set (http://192.168.0.2:8080/axon/js/Property.js?bust=1423077783575:62:16)
    at MovableShape.Object.defineProperty.set [as animating] (http://192.168.0.2:8080/axon/js/PropertySet.js?bust=1423077783575:133:43)
    at MovableShape.inherit.setDestination (http://192.168.0.2:8080/area-builder/js/common/model/MovableShape.js?bust=1423077783575:110:24)
```

**Comment by jonathanolson** [comment]
Still looks to be present in current automated testing.

**Comment by jbphet** [comment]
This issue was occurring when a shape was animating into position on the shape placement board and, before it arrived, other pieces were removed from the board that caused the incoming piece to be 'orphaned' (i.e. in a location that isn't a part of a single contiguous set of tiles) before it arrived.  When it arrived, it was immediately sent back to the origin, which is correct, but other listeners where trying to decompose the shape into individual unit squares.  Code has been added to handle this situation more smoothly, see the commits for details.

**Comment by jbphet** [comment]
This has been fixed, and fuzz testing on two machines for 1.5 hrs on each resulted in no occurrences of this error or any others.  The test server also gives it a green light now.  Closing.
